credsweeper/scanner/scan_type/pem_key_pattern.py

```diff
@@ -38,35 +40,39 @@ def run(cls, config: Config, rule: Rule, target: AnalysisTarget) -> Optional[Can
         """
         assert rule.pattern_type == rule.PEM_KEY_PATTERN, \
             "Rules provided to PemKeyPattern.run should have pattern_type equal to PEM_KEY_PATTERN"
-
-        if cls.is_pem_key(target.lines[target.line_num:], config):
+        global pem_pattern_check
+        if not pem_pattern_check:
+            pem_pattern_check = ValuePemPatternCheck(config)
+        if cls.is_pem_key(target.lines[target.line_num:], target.line_num):
             return cls._get_candidate(config, rule, target)
 
         return None
 
     @classmethod
-    def is_pem_key(cls, lines: List[str], config: Config) -> bool:
+    def is_pem_key(cls, lines: List[str], start_line: int) -> bool:
         """Check if provided lines is a PEM key.
 
         Args:
             lines: Lines to be checked
+            start_line: first line where the data start
 
         Return:
             Boolean. True if PEM key, False otherwise
 
         """
+        limit_line = start_line + 190
         lines = cls.strip_lines(lines)
         lines = cls.remove_leading_config_lines(lines)
         key_data = ""
         for line_num, line in enumerate(lines):
-            if line_num >= 190:
+            if line_num >= limit_line:
```

Comment:
Why the `start_line` parameter is needed?
Comparing `line_num` is just compare not to over `190` lines, not the original line number(`start_line` + `190`).
Please check.

Reply:
Refactored with limit of 190 lines before to skip extra copies.

Reply:
I commented few.. please check below.